app/src/main/java/com/google/android/stardroid/search/PrefixStore.java

```diff
@@ -27,39 +23,61 @@
  *
  */
 public class PrefixStore {
-  // This is a naive implementation that simply stores all the prefixes in a
-  // hashmap.  A better solution would be to use a trie.
-  // TODO(johntaylor): reimplement this, probably as a trie.
```

Comment:
Fantastic.  Thanks for addressing this long-forgotten TODO.  I'd forgotten we'd done it this way - we must have really been under pressure to ship this.